src/aihwkit/simulator/tiles/base.py

```diff
@@ -526,7 +542,8 @@ def __setstate__(self, state: Dict) -> None:
             d_size = self.out_size
 
             # Recreate the tile.
-            self.tile = self._create_simulator_tile(x_size, d_size, self.rpu_config)
+            self.tile = self._recreate_simulator_tile(x_size, d_size, self.rpu_config)
```

Comment:
Ok, that might be a solution, however, does the "load_rpu_config" mechanism works with that change still?

I still think that there is just some issue with this call here: https://github.com/IBM/aihwkit/blob/cfccd18db2f56957304026dc9e79da0b5611bb5b/src/aihwkit/simulator/tiles/module.py#L116

It might be that there is some tricky inheritance issue, maybe introduced with a new torch version.

Reply:
But, OK, probably the issue is that torch interferes with the recreation because in case of the TorchTIle the SimulatorTile is also a Module so torch messes around here as well. 
https://github.com/IBM/aihwkit/blob/master/src/aihwkit/simulator/tiles/torch_tile.py#L37
The recreate solution might be OK and quite clean if everything works otherwise